app/services/duplicate_profile_resolver.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+# A service class to merge duplicate Profile objects
+class DuplicateProfileResolver
+  class << self
+    def run!
+      @profiles = nil
```

Comment:
Note that this is a class variable. Subsequent runs of this resolver on the same thread will cache whatever is stored here, so we need to clear it. This is especially true of the tests.